frontend: Automatically load Tectonic license / pull secret if they exist on disk

kyoto opened the pull request:
Replaces the `/containerlinux/images/amis` API endpoint with a `/tectonic/facts`, which includes all of the initial data needed by the GUI.

Also removes the `textarea`s for the license and pull secret inputs.

The license file / pull secret directory will be made configurable via a command line option / environment variable in a future PR.

Values in state override those returned by `/tectonic/facts`, so
- Uploading a progress file can override any license / pull secret files on disk
- Uploading a different file, then refreshing the page will keep the uploaded file

### AWS
![screenshot-2](https://user-images.githubusercontent.com/460802/31366320-47dab9f0-adab-11e7-8246-589436cf7b63.png)

### Bare Metal
![screenshot-1](https://user-images.githubusercontent.com/460802/31366324-4b373f88-adab-11e7-8bfe-2b787ddc4300.png)

ggreer commented on installer/frontend/components/tectonic-license.jsx:
```diff
@@ -1,91 +1,124 @@
-    </div>
+const FileInput = ({id, onValue}) => {
+  const upload = e => {
+    readFile(e.target.files.item(0))
```
This brings the number of file upload parsers to 3. There's one for restoring from a progress file, another for SSH keys (previously also used for license & pull secret), and now this. Their functionality and purposes seem to overlap a decent amount, but I'm not sure how easy it would be to generalize & dedupe the code.

I think this change is fine, but it'd be a good target for cleanup in the future.

kyoto replied:
Thanks, I'll try to clean this up in a future PR.

kyoto commented:
@ggreer I added a commit to make it to only query the available AMIs if platform `aws-tf` is enabled.

squat commented on installer/api/handlers_tectonic.go:
```diff
@@ -342,3 +343,43 @@ func tectonicKubeconfigHandler(w http.ResponseWriter, req *http.Request, ctx *Co
 	return nil
 }
+
+// Get list of available Container Linux AMIs as well as the Tectonic license
```
Go convention, and requirement for generating docs, is for function comments to begin with the function name. In this case it should look like:
```go
// tectonicFactsHandler gets a listof available Container Linux AMIs as well as the Tectonic
// license and pull secret if they exist.
```

kyoto replied:
Fixed

squat commented on installer/api/handlers_tectonic.go:
```diff
@@ -342,3 +343,43 @@ func tectonicKubeconfigHandler(w http.ResponseWriter, req *http.Request, ctx *Co
+
+	license, err := ioutil.ReadFile(filepath.Join(filepath.Dir(ex), "license.txt"))
+	if err != nil {
+		license = nil
```
consider logging this error as a warning

kyoto replied:
Decided to leave as is, but can add a warning in the future if people feel it would be helpful.

squat commented on installer/api/handlers_tectonic.go:
```diff
@@ -342,3 +343,43 @@ func tectonicKubeconfigHandler(w http.ResponseWriter, req *http.Request, ctx *Co
+
+	pullSecret, err := ioutil.ReadFile(filepath.Join(filepath.Dir(ex), "pull_secret.json"))
+	if err != nil {
+		pullSecret = nil
```
same

kyoto replied:
same

squat commented on installer/api/handlers_tectonic.go:
```diff
@@ -342,3 +343,43 @@ func tectonicKubeconfigHandler(w http.ResponseWriter, req *http.Request, ctx *Co
+		pullSecret = nil
+	}
+
+	type Response struct {
```
we reserve uppercase type names for exported types. since this is only relevant for this one function, lets make it lowercase.

kyoto replied:
Fixed

kyoto commented:
@squat Addressed your comments. I changed the commit messages to `installer:`. I prefer not to break up the commits since the backend and frontend changes depend on each other.
